Core/GameEngine/Source/Common/System/MiniDumper.cpp

```diff
@@ -0,0 +1,725 @@
+/*
+**	Command & Conquer Generals Zero Hour(tm)
+**	Copyright 2025 TheSuperHackers
+**
+**	This program is free software: you can redistribute it and/or modify
+**	it under the terms of the GNU General Public License as published by
+**	the Free Software Foundation, either version 3 of the License, or
+**	(at your option) any later version.
+**
+**	This program is distributed in the hope that it will be useful,
+**	but WITHOUT ANY WARRANTY; without even the implied warranty of
+**	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+**	GNU General Public License for more details.
+**
+**	You should have received a copy of the GNU General Public License
+**	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "PreRTS.h"	// This must go first in EVERY cpp file in the GameEngine
+
+#ifdef RTS_ENABLE_CRASHDUMP
+#include "Common/MiniDumper.h"
+#include "Common/GameMemory.h"
+#include "gitinfo.h"
+
+// Globals for storing the pointer to the exception
+_EXCEPTION_POINTERS* g_dumpException = NULL;
+DWORD g_dumpExceptionThreadId = 0;
+
+MiniDumper* TheMiniDumper = NULL;
+
+// Globals containing state about the current exception that's used for context in the mini dump.
+// These are populated by MiniDumper::DumpingExceptionFilter to store a copy of the exception in case it goes out of scope
+_EXCEPTION_POINTERS g_exceptionPointers = { 0 };
+EXCEPTION_RECORD g_exceptionRecord = { 0 };
+CONTEXT g_exceptionContext = { 0 };
+
+void MiniDumper::initMiniDumper(const AsciiString& userDirPath)
+{
+	DEBUG_ASSERTCRASH(TheMiniDumper == NULL, ("MiniDumper::initMiniDumper called on already created instance"));
+
+	// Use placement new on the process heap so TheMiniDumper is placed outside the MemoryPoolFactory managed area.
+	// If the crash is due to corrupted MemoryPoolFactory structures, try to mitigate the chances of MiniDumper memory also being corrupted
+	TheMiniDumper = new (::HeapAlloc(::GetProcessHeap(), HEAP_GENERATE_EXCEPTIONS, sizeof(MiniDumper))) MiniDumper;
```

Comment:
After #1066 is merged, you can also use `sysAllocateDoNotZero` / `sysFree` for it.